### ISSUE TITLE
signing a single message

### ISSUE THREAD

**Issue by davidleomay** [DeFiCh/wallet#688]
When trying to sign a message with the newest version of the Light Wallet with `const signature = await account.sign(hash)`, an exception with the message `No UI attached for passphrase prompting` is thrown. I followed the hint in `WalletNodeProvider` to `TransactionAuthorization`, but couldn't figure out how to sign a single message (not a transaction).
Is anyone willing to give a hint on how to do the signing with the newest version?

**Comment by ivan-zynesis** [comment]
> When trying to sign a message with the newest version (v0.7.0) of the wallet with `const signature = await account.sign(hash)`, an exception with the message `No UI attached for passphrase prompting` is thrown.

Hi @davidleomay I am guessing this happened in your code thus the error message `No UI attached for passphrase prompting`
```ts
const accountFromWalletContext = useWallet()
const signer = (accountFromTransactionAuth) => {
  // accountFromWalletContext.sign() // this will fail
  accountFromTransactionAuth.sign()
}
```

> To integrate the [Fiat exchange service](https://fiat2defi.ch/) into the Light Wallet we need the functionality to sign a simple message. 

This is actually a different use case, signing any message for other integration). As you can see our code setup a flow to perform a signing which output `CTransactionSegwit`. So you have to setup another "action" which will proceed with your desired implementation after the passphrase collected from user.

```ts
const provider = MnemonicEncrypted.initProvider(providerData, network, { prompt: onPrompt })
const wallet = initJellyfishWallet(provider, network, client)
wallet.get(0).sign(<your custom message here>)
```

**Comment by davidleomay** [comment]
Hi @ivan-zynesis, thanks for your explanation, that helped very much. The `TransactionAuthorization` contains all the logic to enter the pin code, but it is specific for signing transactions. If we now want to sign a message, we would need to copy/re-implement most of the UI and logic to enter and validate the pin code. Is that correct?
Are there any plans on your side to extract this common logic? Or is there a simpler way to reuse it?

**Comment by ivan-zynesis** [comment]
Yes, it is possible to reuse the entire pin input UI + logic implementation (thus it handle pin attempt counter, retry logic etc). Assuming your codebase still have very similar structure as this repo. Following snippet should fit your use case.

```ts
   const auth: Authentication<Buffer> = {
      consume: async passphrase => {
        // quickly instantiate a wallet to sign your message
        const provider = MnemonicEncrypted.initProvider(providerData, network, { prompt: () => passphrase }) // feed collected value directly
        const wallet = initJellyfishWallet(provider, network, client)

        // it will throw error here and direct to next attempt if passphrase invalid
        return await wallet.get(0).sign(<your custom message here>)
      },
      onAuthenticated: async (signature: Buffer) => {
        // TADA!
        continueWithYourIntegration(signature)
      },
      onError: e => Logging.error(e),
      message: 'todo',
      loading: 'todo'
    }
    dispatch(authentication.actions.prompt(auth))
```

https://github.com/DeFiCh/wallet/blob/main/app/screens/AppNavigator/screens/Settings/SettingsScreen.tsx#L36-L45
